### PR TITLE
Add details to stream error response

### PR DESCRIPTION
**Pull request by johanbrandhorst**
Add the gRPC status details to the streamed response.

This is built on top of https://github.com/grpc-ecosystem/grpc-gateway/pull/560, but I can close that one if we want to merge these together instead.

### CODE REVIEW

**Comment by johanbrandhorst** [comment]
Probably easiest to merge this and close #560 if we want both of these changes.

File: runtime/handler.go [hunk 1]
```diff
@@ -168,17 +168,15 @@ func handleForwardResponseStreamError(wroteHeader bool, marshaler Marshaler, w h
 
 func streamChunk(result proto.Message, err error) map[string]proto.Message {
 	if err != nil {
-		grpcCode := codes.Unknown
-		if s, ok := status.FromError(err); ok {
-			grpcCode = s.Code()
-		}
-		httpCode := HTTPStatusFromCode(grpcCode)
+		s, _ := status.FromError(err)
```
Comment (srenatus) [inline runtime/handler.go]: Hmm. I know it should never happen, but seeing the second return ignored is weird, too. Any thoughts about something like
```go
s, ok := status.FromError(err)
if !ok {
  panic("err is no status")
}
```

Comment (johanbrandhorst) [reply runtime/handler.go]: I wanted to make use of the new status handling in [gRPC-Go 1.10](https://github.com/grpc/grpc-go/commit/d50734d1d6ca477a72646f3022216ec39639f4cd) which is equivalent to the old handling, but I accept gRPC-Gateway can't enforce that users have the latest gRPC version. I will change it back.

Comment (srenatus) [reply runtime/handler.go]: Oh that's a lovely function. Thanks for the pointer.

**Comment by johanbrandhorst** [comment]
I hate to bump this but I think it's a pretty straightforward, backwards compatible change with a new test case and I'd _really_ like to have this merged ASAP. Please advise if I can speed things up somehow.

**Comment by achew22** [comment]
I just reviewed this and I think it is good to go. Marking as approved. Can you please close any other PRs that this obviates and then I'll merge it (probably tomorrow morning/early afternoon US time)

**Comment by johanbrandhorst** [comment]
Closed #560 and #564.

**Comment by achew22** [comment]
Today was a disaster. Sorry for the slow merge. Merging now

**Comment by johanbrandhorst** [comment]
Any chance we could have a new release with this? 1.4.0?

**Pull request merged**